### PR TITLE
Make configurable basic UI WebSocket request path and move UI to a template file

### PR DESCRIPTION
**Pull request by maximthomas**

### CODE REVIEW

File: cmd/echo-server/main.go [hunk 1]
```diff
@@ -100,9 +102,7 @@ func handler(wr http.ResponseWriter, req *http.Request) {
 	if websocket.IsWebSocketUpgrade(req) {
 		serveWebSocket(wr, req, sendServerHostname)
 	} else if req.URL.Path == "/.ws" {
-		wr.Header().Add("Content-Type", "text/html")
-		wr.WriteHeader(200)
-		io.WriteString(wr, websocketHTML) // nolint:errcheck
+		serveFromntend(wr)
```
Comment (onedr0p) [inline cmd/echo-server/main.go:105]:
```suggestion
		serveFrontend(wr)
```

File: cmd/echo-server/main.go [hunk 2]
```diff
@@ -159,6 +159,30 @@ func serveWebSocket(wr http.ResponseWriter, req *http.Request, sendServerHostnam
 	}
 }
 
+//go:embed "html"
+var files embed.FS
+
+func serveFromntend(wr http.ResponseWriter) {
```
Comment (onedr0p) [inline cmd/echo-server/main.go:165]:
```suggestion
func serveFrontend(wr http.ResponseWriter) {
```

**Review by onedr0p — CHANGES_REQUESTED** [review]

**Review by onedr0p — APPROVED** [review]

**Comment by jmalloc** [comment]
Thanks folks. I am happy to merge this, but out of curiosity, would your particular use case also be solved if the frontend's `wss://` URL were to use to the same path at which the frontend is served?

**Comment by maximthomas** [comment]
Hi @jmalloc, 
I've encountered a problem where the echo-server should be deployed behind a reverse proxy. There is another application behind the same proxy that should be accessible using the root path. So, I proposed the option to eliminate the conflict.

**Comment by jmalloc** [comment]
Thanks again! Would PR https://github.com/jmalloc/echo-server/pull/35 also solve your use case?

**Comment by maximthomas** [comment]
@jmalloc Not quite. PR #35 is about ws and sse endpoints, and the current PR is about the UI

**Comment by jmalloc** [comment]
I've released [v0.3.7](https://github.com/jmalloc/echo-server/releases/v0.3.7), which adds the `WEBSOCKET_ROOT` environment variable, which is very similar to the `FRONTEND_WS_PATH` variable added by this PR. It combines the behaviour of this PR with that of #35, wherein the path that the `.ws` UI is served from is also added to the websocket URL. This should also fix #24.

**Pull request merged**